src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,29 +413,94 @@ describe(df)
 ```
 
 """
-StatsBase.describe(df::AbstractDataFrame) = describe(stdout, df)
-function StatsBase.describe(io, df::AbstractDataFrame)
-    for (name, col) in eachcol(df)
-        println(io, name)
-        describe(io, col)
-        println(io, )
+StatsBase.describe(df::AbstractDataFrame; kwargs...) = describe(stdout, df; kwargs...)
+function StatsBase.describe(io, df::AbstractDataFrame; stats = [:mean, :min, :median, :max, :nmissing, :eltype])
+    # Check that people don't specify the wrong fields. 
+    allowed_fields = [:mean, :std, :min, :q25, :median, :q75, :max, :nunique, :nmissing, :eltype] 
+    if !issubset(stats, allowed_fields) 
+        disallowed_fields = setdiff(stats, allowed_fields)
+        not_allowed = "Field(s) not allowed: $disallowed_fields. "
+        allowed = "Allowed fields are: $allowed_fields."
+        throw(ArgumentError(not_allowed * allowed)) 
     end
-end
 
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{<:Real})
```

Comment:
Use a different name for the `col` argument, since it triggers an error in Julia 0.7 (apparently due to the conflict with the variable in the parent function).

Reply:
Actually it's totally unrelated to the existence of a `col` variable in the function. Can you move the definitions of `get_stats` out of the function for now to work around the bug in Julia 0.7?

Reply:
I moved all of them out of the `describe` function but kept the `col` argument name.